src/System.Linq/src/System/Linq/First.cs

```diff
@@ -10,47 +10,53 @@ public static partial class Enumerable
     {
         public static TSource First<TSource>(this IEnumerable<TSource> source)
         {
+            bool found;
+            TSource first = source.TryGetFirst(out found);
+            
+            if (!found)
+            {
+                throw Error.NoElements();
+            }
+
+            return first;
+        }
+
+        public static TSource First<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate)
```

Comment:
@stephentoub Good idea. I've done this for the predicate overloads too. There are now more than 170 deleted lines :tada: